desdeo/mcdm/pareto_navigator.py

```diff
@@ -0,0 +1,271 @@
+        current_solution (dict[str, float]): The currently navigated point.
+        alpha (float): Step size. Between 0 and 1.
+        matrix_a (np.ndarray): The A' matrix.
+        b (np.ndarray): The b vector.
+
+    Returns:
+        dict[str, float]: The next solution.
+    """
+    z = objective_dict_to_numpy_array(problem, current_solution)
+    k = len(z)
+    d = objective_dict_to_numpy_array(problem, search_direction)
+
+    q = z + alpha * d
+    q = np.reshape(q, ((k, 1)))
+
+    b_new = np.append(q, b)
+
+    ideal = objective_dict_to_numpy_array(problem, get_ideal_dict(problem))
+    nadir = objective_dict_to_numpy_array(problem, get_nadir_dict(problem))
+
+    c = np.array([1] + k * [0])
+
+    obj_bounds = np.stack((ideal, nadir))
+    bounds = [(None, None)]
+    for x, y in obj_bounds.T:
+        bounds.append((x, y))
+
+    z_new = linprog(c=c, A_ub=matrix_a, b_ub=b_new, bounds=bounds)
+    if z_new["success"]:
+        return numpy_array_to_objective_dict(problem, z_new["x"][1:])
+    return current_solution # should raise an exception instead
+
+def calculate_all_solutions(
+    problem: Problem,
+    current_solution: dict[str, float],
+    alpha: float,
+    num_solutions: int,
+    pref_info: dict
+) -> list[dict[str, float]]:
+    """Performs a set number of steps in the current direction.
+
+    Args:
+        problem (Problem): The problem being solved.
+        current_solution (dict[str, float]): The current solution.
+        alpha (float): Step size. Between 0 and 1.
+        num_solutions (int): Number of solutions calculated.
+        pref_info (dict): Preference information. Either "reference_point" or "classification".
+
+    Returns:
+        list[dict[str, float]]: A list of the computed solutions.
+    """
+    solution = current_solution
+
+    # check if the preference information is given as a reference point or as classification
+    # and calculate the search direction based on the preference information
+    if "reference_point" in pref_info:
+        reference_point = numpy_array_to_objective_dict(problem, pref_info["reference_point"])
+        d = calculate_search_direction(problem, reference_point, current_solution)
+    elif "classification" in pref_info:
+        reference_point = classification_to_reference_point(problem, pref_info, solution)
+        d = calculate_search_direction(problem, reference_point, solution)
+
+    # the A matrix and b vector from the polyhedral set equation
+    matrix_a, b = get_polyhedral_set(problem)
+
+    # the A' matrix from the linear parametric programming problem
+    matrix_a_new = construct_matrix_a(problem, matrix_a)
+
+    solutions: list[dict[str, float]] = []
+    while len(solutions) < num_solutions:
+        solution = calculate_next_solution(problem, d, solution, alpha, matrix_a_new, b)
+        solutions.append(solution)
+    return solutions
+
+# Testing
+if __name__ == "__main__":
+    from desdeo.problem import pareto_navigator_test_problem
```

Comment:
Can you move these tests to the folder `tests`? It would be nice if there are some asserts as well, but the bare minimum should be that the code runs at least in some case without errors, like is done here now.

Reply:
Tein tuosta testauksesta oman tiedostonsa ja lisäsin sinne pari assertia ja ainakin näillä meni testit läpi.